crowbar_framework/app/models/ceph_service.rb

```diff
@@ -289,6 +289,20 @@ def validate_proposal_after_save proposal
       end
     end
 
+    nodes_without_suitable_drives = proposal["deployment"][@bc_name]["elements"]["ceph-osd"].select do |node_name|
+      node = NodeObject.find_node_by_name(node_name)
+      if node.nil?
+          false
+      else
+          candidate_disks_count = node.unclaimed_physical_drives.length + node.physical_drives.select { |d, data| node.disk_owner(node.unique_device_for(d)) == 'Ceph' }.length
```

Comment:
Please split the part about finding ceph disks on another line with a variable. This will make the code more readable.